### PR TITLE
feat: add watchlist endpoint implementation

### PR DESCRIPTION
**Pull request by cksidharthan**
Alpaca Documentation - https://alpaca.markets/docs/api-references/trading-api/watchlist/

Endpoints Added:
- GET - `/v2/watchlists` - Get all watchlists
- GET - `/v2/watchlists/{watchlist_id}` - Get a watchlist by id
- POST - `/v2/watchlists` - Create Watchlist
- POST - `/v2/watchlists/{watchlist_id}` - Add asset to watchlist
- PUT - `/v2/watchlists/{watchlist_id}` - Update watchlist by id
- DELETE - `/v2/watchlists/{watchlist_id}` - Delete a watchlist by id
- DELETE - `/v2/watchlists/{watchlist_id}/{symbol}` - Remove a symbol from watchlist

### CODE REVIEW

File: alpaca/rest.go [hunk 1]
```diff
@@ -772,6 +779,126 @@ func (c *client) GetAnnouncement(announcementID string) (*Announcement, error) {
...
+	watchlist := &Watchlist{}
+
+	if err = unmarshal(resp, watchlist); err != nil {
+		return nil, err
+	}
+
+	return watchlist, nil
+}
+
+func (c *client) RemoveSymbolFromWatchlist(watchlistID string, symbol string) error {
```
Comment (gnvk) [inline alpaca/rest.go:882]: Any reason why `AddAssetToWatchlist` and `RemoveSymbolFromWatchlist` are asymmetrical? One uses the word "asset" and a request, the other uses "symbol", as an argument. I would pick one. The second is easier to read and use, but the first is more expandable, if any other parameters are ever added to the endpoint.

File: alpaca/rest_test.go [hunk 2]
```diff
@@ -443,6 +443,192 @@ func TestClient_GetAnnouncement(t *testing.T) {
...
+
+		return &http.Response{
+			Body: genBody(watchlists),
+		}, nil
+	}
+
+	watchlists, err := c.GetWatchlists()
+	require.NoError(t, err)
+	require.Len(t, watchlists, 1)
+}
```
Comment (gnvk) [inline alpaca/rest_test.go:477]: I would also assert that at least some of the fields of the watchlist actually matches.

File: alpaca/rest_test.go [hunk 3]
```diff
@@ -443,6 +443,192 @@ func TestClient_GetAnnouncement(t *testing.T) {
...
+
+func TestClient_UpdateWatchlist(t *testing.T) {
+	c := testClient()
+	// successful
+	c.do = func(c *client, req *http.Request) (*http.Response, error) {
+		assert.Equal(t, "api.alpaca.markets", req.URL.Host)
+		assert.Equal(t, "/v2/watchlists/123", req.URL.Path)
+		assert.Equal(t, "PUT", req.Method)
+
```
Comment (gnvk) [inline alpaca/rest_test.go:551]: I would also check the request body here.

File: alpaca/rest_test.go [hunk 4]
```diff
@@ -443,6 +443,192 @@ func TestClient_GetAnnouncement(t *testing.T) {
...
+}
+
+func TestClient_AddToWatchlist(t *testing.T) {
+	c := testClient()
+	// successful
+	c.do = func(c *client, req *http.Request) (*http.Response, error) {
+		assert.Equal(t, "api.alpaca.markets", req.URL.Host)
+		assert.Equal(t, "/v2/watchlists/123", req.URL.Path)
+		assert.Equal(t, "POST", req.Method)
+
```
Comment (gnvk) [inline alpaca/rest_test.go:602]: Same here.

**Comment by gnvk** [comment]
@cksidharthan Thank you for this PR! It's very good, I only had a few suggestions.

File: alpaca/rest.go (hunk 1 above)
Comment (cksidharthan) [reply alpaca/rest.go:882]: I was following the alpaca documentation Subheading of the api endpoint. :) Now I have replaced `AddAssetToWatchlist` with `AddSymbolToWatchlist` now in the latest commit :)

File: alpaca/rest_test.go (hunk 2 above)
Comment (cksidharthan) [reply alpaca/rest_test.go:477]: I have added the fix in the latest commit 😄

File: alpaca/rest_test.go (hunk 3 above)
Comment (cksidharthan) [reply alpaca/rest_test.go:559]: I have added the fix in latest commit 😄

File: alpaca/rest_test.go (hunk 4 above)
Comment (cksidharthan) [reply alpaca/rest_test.go:614]: I have added the fix in the latest commit 😄

File: alpaca/rest.go (hunk 1 above)
Comment (gnvk) [reply alpaca/rest.go:882]: Sorry about being nit-picky here, but now the `AddSymbolToWatchlist` uses `AddSymbolToWatchlistRequest` while `RemoveSymbolFromWatchlist` simply uses the `symbol` as a parameter. Please unify them.

Comment (cksidharthan) [reply alpaca/rest.go]: Sure. I've fixed it now in the latest commit 😄

**Review by gnvk — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**